flink-kubernetes-operator/pom.xml

```diff
@@ -61,6 +61,12 @@ under the License.
             <groupId>org.apache.flink</groupId>
             <artifactId>flink-clients_${scala.version}</artifactId>
             <version>${flink.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.flink</groupId>
+                    <artifactId>flink-scala_${scala.version}</artifactId>
+                </exclusion>
+            </exclusions>
```

Comment:
What is the reason for excluding this?

Reply:
All the interfaces we are using is scala free, so do we really need to bundle scala in the uber jar?

Reply:
why would it be in the client if it's not using it?

Reply:
I dont actually know if we are going to hit any issues here, just playing the devil's advocate :)

Reply:
safest would be for now is to keep it included

Reply:
The scala dependency was introduced by `flink-streaming-java_2.12`. I remember the root cause is about akka.

Reply:
I am almost sure we are not using scala classes in the `flink-kubernetes-operator`. That is the reason I excluded it to reduce the jar size. But I could add it back if you are concerned about this.

Reply:
I am not concerned should be ok :)

Reply:
streaming-java had a transitive dependency on flink-scala in 1.14-, which was removed in 1.15.0. To keep the dependency graph identical we added an explicit dependency on flink-scala.
If you don't run any application within the operator itself then you shouldn't need it.

That said you also shouldn't need a bunch of other stuff, like the QS client, rpc-akka-loader, asm, flink-optimizer etc etc.